engine/access/ingestion2/collection_syncer.go

```diff
@@ -405,17 +424,14 @@ func (s *CollectionSyncer) findLowestBlockHeightWithMissingCollections(
 }
 
 // OnCollectionDownloaded indexes and persists a downloaded collection.
-// This is a callback intended to be used with the requester engine.
+// This function is a callback intended to be used by the requester engine.
+//
+// Note: You might wonder what happens if the ingestion engine crashes,
+// but the requester engine continues to invoke this callback and writes
+// to a closed channel. In our current design, we intentionally ignore such cases.
+//
+// If any engine crashes, we treat it as a sign of possible state corruption,
+// and the node is expected to restart. Thus, this edge case is not handled explicitly.
 func (s *CollectionSyncer) OnCollectionDownloaded(_ flow.Identifier, entity flow.Entity) {
-	collection, ok := entity.(*flow.Collection)
-	if !ok {
-		s.logger.Error().Msgf("invalid entity type (%T)", entity)
-		return
-	}
-
-	err := indexer.IndexCollection(collection, s.collections, s.transactions, s.logger, s.collectionExecutedMetric)
-	if err != nil {
-		s.logger.Error().Err(err).Msg("could not index collection after it has been downloaded")
-		return
-	}
+	s.pendingCollections <- entity
```

Comment:
I think this should use a message handler instead of a channel. this allows us to avoid blocking the caller if the main loop is busy, which can happen since it's used for period work as well.